File: youlexuan-manager-web/src/main/java/com/offcn/sellergoods/controller/BrandController.java
package com.offcn.sellergoods.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.offcn.entity.PageResult;
import com.offcn.entity.Result;
import com.offcn.pojo.TbBrand;
import com.offcn.sellergoods.service.BrandService;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/brand")
public class BrandController {

    @Reference(timeout = 5000)
    BrandService brandService;

    @RequestMapping("/findAll")   // brand/findAll.do
    public List<TbBrand> findAll(){
        return brandService.findAll();
    }

    @RequestMapping("/findPage")
    public PageResult findPage(int pageNum,int pageSize){
        return brandService.findPage(pageNum,pageSize);
    }

    @RequestMapping("/add")
    public Result add(@RequestBody TbBrand brand){ // 前台传过来的json格式： {id:1,name:'aaaaa'} ==>RequestBody ==> TbBrand brand
        Result result = brandService.add(brand);
        return  result;
    }

    @RequestMapping("/update")
    public  Result update(@RequestBody TbBrand brand){
        Result update = brandService.update(brand);
        return update;
    }

    @RequestMapping("/findOne")
    public TbBrand findOne(Long id){
        return brandService.findById(id);
    }

    @RequestMapping("/delete")
    public Result delete(Long[] ids){
        try {
            brandService.delete(ids);
            return  new Result(true,"删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"删除失败");
        }
    }


    /*带条件的分页查询*/
    @RequestMapping("/search")
    public PageResult search(int pageNum,int pageSize,@RequestBody TbBrand searchEntity){
//        aaa();
        return brandService.findPage(pageNum,pageSize,searchEntity);
    }

    public List<Map> getBrandList(){
//        [{id:1,text:''},{},{}]  // select id, name as 'text' from tb_brand;
        List<TbBrand> list = brandService.findAll();

        List<Map> result = new ArrayList<>();

        for (TbBrand brand : list) {
            Map map = new HashMap();
            map.put("id",brand.getId());
            map.put("text",brand.getName());
            result.add(map);
        }
        return result;
    }

}

File: youlexuan-manager-web/src/main/webapp/js/controller/brandController.js
/*控制层*/
/*brandController注入brandService*/
app.controller('brandController',function ($scope,$http,brandService,$controller) {

    //继承baseController
    $controller('baseController',{$scope:$scope});

    $scope.findAll = function () {
        // $http.get("../brand/findAll.do").success(function (resp) {
        brandService.findAll().success(function (resp) {
            $scope.brandList = resp;//resp返回的数据
        });
    };

    $scope.findPage = function(pageNum,pageSize){
        // $http.get("../brand/findPage.do?pageNum="+pageNum+"&pageSize="+pageSize).success(function (resp) {
        brandService.findPage(pageNum,pageSize).success(function (resp) {
            //resp == PageResult
            $scope.paginationConf.totalItems = resp.total;//总记录数，赋值给paginationConf.totalItems
            $scope.brandList = resp.rows;//数据集合
        });
    };




    //修改和添加，都要执行save方法，所有我么在save中判断，是需要完成添加还是修改，根据id
    $scope.save = function () {

        if($scope.entity.id==null){
            //添加
            // $http.post("../brand/add.do",$scope.entity).success(function (resp) {
            brandService.add($scope.entity).success(function (resp) {
                //resp == Result (success,message)
                if(resp.success){
                    //添加成功，重新加载数据
                    $scope.reloadList();
                    //添加成功，entity清空
                    $scope.entity = {};
                }else {
                    alert(resp.message)
                }
            });
        }else{
            //entity有id，完成修改
            // $http.post("../brand/update.do",$scope.entity).success(function (resp) {
            brandService.update($scope.entity).success(function (resp) {
                if(resp.success){
                    $scope.reloadList();
                    $scope.entity = {};
                }else{
                    alert(resp.message);
                }
            });
        }
    }

    $scope.findOne = function (id) {
        // $http.get("../brand/findOne.do?id="+id).success(function (resp) {
        brandService.findOne(id).success(function (resp) {
            //resp == 品牌对象
            $scope.entity = resp;
        });
    }



    $scope.delete = function () {
        // $http.get("../brand/delete.do?ids="+$scope.selectIds).success(function (resp) {
        brandService.delete($scope.selectIds).success(function (resp) {
            if(resp.success){
                $scope.reloadList();
                $scope.selectIds = [];
            }else{
                alert(resp.message)
            }
        });
    };

    //初始化searchEntity
    $scope.searchEntity = {};

    $scope.search = function (pageNum,pageSize) {
        // $http.post("../brand/search.do?pageNum="+pageNum+"&pageSize="+pageSize,$scope.searchEntity).success(function (resp) {
        brandService.search(pageNum,pageSize,$scope.searchEntity).success(function (resp) {
            // resp = PageResult
            $scope.paginationConf.totalItems = resp.total;//总记录数，赋值给paginationConf.totalItems
            $scope.brandList = resp.rows;//数据集合
        });
    };






});